datasync/chaindatafetcher/kafka/kafka_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package kafka_test
+
+import (
+	"encoding/json"
+	"math/rand"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/klaytn/klaytn/datasync/chaindatafetcher/kafka"
+	"github.com/klaytn/klaytn/datasync/chaindatafetcher/types"
+	"github.com/stretchr/testify/suite"
+)
+
+type kafkaData struct {
+	Name   string `json:"name"`
+	Data   string `json:"data"`
+	Number int    `json:"number"`
+	IsGood bool   `json:"isGood"`
+}
+
+type KafkaSuite struct {
+	suite.Suite
```

Comment:
Does Suite makes it enable to run test codes of singleton pattern?
If so, we should apply them to other place, too. :)

Reply:
Hmm.. `SetupSuite` is run just before each test function, so I created a new kafka struct every test.

Reply:
@jeongkyun-oh @winnie-byun AFAIK, `SetupSuite` and `TearDownSuite` are called only once for the given suite. If you want to call function before/after each test, we should use `SetupTest` or `TearDownTest`. 


> Does `Suite` makes it enable to run test codes of singleton pattern?
> If so, we should apply them to other place, too. :)

@winnie-byun Something similar. `suite` package helps you avoiding repetitive resource allocation/deallocation. (and others, too!)